analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala

```diff
@@ -190,16 +254,30 @@ class JobAPIService @Inject()(postgresDBUtil: PostgresDBUtil) extends Actor  {
     val key = Array(tag, jobId, requestedBy, requestedChannel, submissionDate).mkString("|")
     MessageDigest.getInstance("MD5").digest(key.getBytes).map("%02X".format(_)).mkString
   }
-  private def _validateRequest(channel: String, eventType: String, from: String, to: String)(implicit config: Config): Map[String, String] = {
+  private def _validateRequest(channel: Option[String], datasetId: String, from: String, to: String, isPublic: Boolean = false)(implicit config: Config): Map[String, String] = {
 
-    APILogger.log("Validating Request", Option(Map("channel" -> channel, "eventType" -> eventType, "from" -> from, "to" -> to)))
+    APILogger.log("Validating Request", Option(Map("channel" -> channel, "datasetId" -> datasetId, "from" -> from, "to" -> to)))
     val days = CommonUtil.getDaysBetween(from, to)
+    val expiryMonths = config.getInt("public.data_exhaust.expiryMonths")
+    val maxInterval = if (isPublic) config.getInt("public.data_exhaust.max.interval.days") else 10
     if (CommonUtil.getPeriod(to) > CommonUtil.getPeriod(CommonUtil.getToday))
       return Map("status" -> "false", "message" -> "'to' should be LESSER OR EQUAL TO today's date..")
     else if (0 > days)
       return Map("status" -> "false", "message" -> "Date range should not be -ve. Please check your 'from' & 'to'")
-    else if (10 < days)
-      return Map("status" -> "false", "message" -> "Date range should be < 10 days")
+    else if (maxInterval < days)
+      return Map("status" -> "false", "message" -> s"Date range should be < $maxInterval days")
+    else if (isPublic && (CommonUtil.getPeriod(from) <  CommonUtil.getPeriod(CommonUtil.dateFormat.print(new DateTime().minusMonths(expiryMonths)))))
+      return Map("status" -> "false", "message" -> s"Date range cannot be older than $expiryMonths months")
     else return Map("status" -> "true")
   }
+
+  def _validateDataset(datasetId: String)(implicit config: Config): Map[String, String] = {
+    val validDatasets = config.getStringList("public.data_exhaust.datasets")
+    if (validDatasets.contains(datasetId)) Map("status" -> "true") else Map("status" -> "false", "message" -> s"Provided dataset is invalid. Please use any one from this list - $validDatasets")
+  }
+
+  def getCDNURL(bucket: String, key: String)(implicit config: Config): String = {
+    val cdnHost = config.getString("cdn.host")
+    cdnHost + bucket + "/" + key
+  }
```

Comment:
CDN url will replace the bucket id as well. Key should be sufficient

Reply:
Fixed CDN url by removing bucket.